lib/rhea-promise/index.ts

```diff
@@ -0,0 +1,674 @@
+      reject(context.receiver.error);
+    }
+
+    receiver.once("receiver_open", onOpen);
+    receiver.once("receiver_close", onClose);
+  });
+}
+
+/**
+ * Closes the amqp receiver.
+ * @param {Receiver} receiver The amqp receiver that needs to be closed.
+ * @return {Promise<void>} Promise<void>
+ * - **Resolves** the promise when rhea emits the "receiver_close" event.
+ * - **Rejects** the promise with an AmqpError when rhea emits the
+ * "receiver_error" event while trying to close an amqp receiver.
+ */
+export function closeReceiver(receiver: any): Promise<void> {
+  if (!receiver || (receiver && typeof receiver !== "object")) {
+    throw new Error("receiver is a required parameter and must be of type 'object'.");
+  }
+
+  return new Promise<void>((resolve, reject) => {
+    if (receiver.is_open()) {
+      function onClose(context: Context): void {
+        receiver.removeListener("receiver_close", onClose);
+        process.nextTick(() => {
+          debug("Resolving the promise as the amqp receiver has been closed.");
+          resolve();
+        });
+      }
+
+      function onError(context: Context): void {
+        receiver.removeListener("receiver_error", onError);
+        debug(`Error occurred while closing amqp receiver.`, context.receiver.error);
+        reject(context.receiver.error);
+      }
+
+      receiver.once("receiver_close", onClose);
+      receiver.once("receiver_error", onError);
+      receiver.close();
+    } else {
+      resolve();
+    }
+  });
+}
+
+/**
+ * Defines the common set of properties that are applicable for a connection, session and a link (sender, receiver).
+ * @interface EntityOptions
+ */
+export interface EntityOptions {
+  /**
+   * @property {any} [desired_capabilities] Extension capabilities the sender can use if the receiver supports them.
+   */
+  desired_capabilities?: any;
+  /**
+   * @property {any} [offered_capabilities] Extension capabilities the sender supports.
+   */
+  offered_capabilities?: any;
+  /**
+   * @property {object} [properties] Properties of the entity (connection, session, link) contain a set of fields
+   * intended to provide more information about the entity.
+   */
+  properties?: { [x: string]: any };
+}
+
+/**
+ * Defines the options that can be provided while creating a connection.
+ * @interface ConnectionOptions
+ * @extends EntityOptions
+ */
+export interface ConnectionOptions extends EntityOptions {
+  /**
+   * @property {string} username - The username.
+   */
+  username: string;
+  /**
+   * @property {string} host - The host to connect to.
+   */
+  host: string;
+  /**
+   * @property {string} hostname - The hostname to connect to.
+   */
+  hostname: string;
+  /**
+   * @property {number} port - The port number (5671 or 5672) at which to connect to.
+   */
+  port: number;
+  /**
+   * @property {string} [transport] - The transport option.
+   */
+  transport?: "tls" | "ssl" | "tcp";
```

Comment:
>"tls" | "ssl" | "tcp" [](start = 14, length = 21)

no ws/wss?

Reply:
This is what I found from the rhea source code over [here](https://github.com/amqp/rhea/blob/master/lib/connection.js#L83-L91) and have updated the type definitions accordingly.
@grs - Any thoughts on Pierre's comment?